core/src/test/java/org/apache/iceberg/rest/requests/TestReportMetricsRequestParser.java

```diff
@@ -0,0 +1,127 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.metrics.ImmutableScanReport;
+import org.apache.iceberg.metrics.MetricsReport;
+import org.apache.iceberg.metrics.ScanReport;
+import org.apache.iceberg.metrics.ScanReport.ScanMetrics;
+import org.apache.iceberg.metrics.ScanReport.ScanMetricsResult;
+import org.apache.iceberg.types.Types;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class TestReportMetricsRequestParser {
+
+  @Test
+  public void nullCheck() {
+    Assertions.assertThatThrownBy(() -> ReportMetricsRequestParser.toJson(null))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid metrics request: null");
+
+    Assertions.assertThatThrownBy(() -> ReportMetricsRequestParser.fromJson((JsonNode) null))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse metrics request from null object");
+  }
+
+  @Test
+  public void missingFields() {
+    Assertions.assertThatThrownBy(() -> ReportMetricsRequestParser.fromJson("{}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing string: report-type");
+
+    Assertions.assertThatThrownBy(
+            () -> ReportMetricsRequestParser.fromJson("{\"report-type\":\"scan-report\"}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing string: table-name");
+
+    Assertions.assertThatThrownBy(
+            () ->
+                ReportMetricsRequestParser.fromJson(
+                    "{\"report-type\":\"scan-report\", \"table-name\" : \"x\"}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing long: snapshot-id");
+  }
+
+  @Test
+  public void invalidReportType() {
+    Assertions.assertThatThrownBy(
+            () -> ReportMetricsRequestParser.fromJson("{\"report-type\":\"invalid\"}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage(
+            "No enum constant org.apache.iceberg.rest.requests.ReportMetricsRequest.ReportType.INVALID");
```

Comment:
Minor: we could replace this with a better error message for the caller. No enum constant doesn't really tell me that I was trying to deserialize an unsupposed report type, invalid.

Reply:
we have a similar pattern in a few other parsers as well. I'll follow up with that

Reply:
I created https://github.com/apache/iceberg/pull/5910 to address this